packages/workspace/src/browser/workspace-service.ts

```diff
@@ -48,7 +48,10 @@ export class WorkspaceService implements FrontendApplicationContribution {
 
     @postConstruct()
     protected async init(): Promise<void> {
-        const rootUri = await this.server.getRoot();
+        let rootUri = await this.server.getRoot();
+        if (rootUri && rootUri.endsWith("/")) {
+            rootUri = rootUri.slice(0, -1);
```

Comment:
`rootUri.slice(0, rootUri.length - 1)` or i am missing something?

Reply:
Code move to toValidRoot. The "/" is preventing the labelPorvider to display the name or the workspace in the navigator view

Reply:
I understand, but code looks strange, why just -1?

Reply:
-1 is only to remove the last char of the uri if it ends with a "/", thats all. Have anything in mind !